docs/streams/getting-started-streams.md

```diff
@@ -323,101 +327,148 @@ else:
+const client = new jsc8(globalUrl);
+await client.login("your@email.com", "password");
+*/
+
+// Variables
+const stream = "streamQuickstart";
+let prefix_text = "";
+const is_local = false; //For a local stream pass this variable as True, or False for a global stream
+
+// Get the right prefix for the stream
+if (is_local) {
+  prefix_text = "c8locals.";
+} else {
+  prefix_text = "c8globals.";
 }
 
-// Create global and local streams.
-async function createStreams() {
-    try{
-      console.log("Creating local stream...");
-      await client.createStream("testStream-local", true);
+async function createMyStream () {
+  let streamName = { "stream-id": "" };
+  if (await client.hasStream(stream, is_local)) {
+    console.log("This stream already exists!");
+    streamName["stream-id"] = prefix_text + stream;
+    console.log(`Old Producer = ${streamName["stream-id"]}`);
+  } else {
+    streamName = await client.createStream(stream, is_local);
+    console.log(`New Producer = ${streamName.result["stream-id"]}`);
+  }
+}
 
-      console.log("Creating global stream...");
-      await client.createStream("testStream-global", false);
+async function sendData () {
+  console.log("\n ------- Publish Messages  ------");
+  const producer = await client.createStreamProducer(stream, is_local);
 
-    } catch(e) {
-      await console.log("Streams could not be created because "+ e);
+  producer.on("open", () => {
+    const input = readline.createInterface({
+      input: process.stdin,
+      output: process.stdout
+    });
+
+    // To repeatedly ask the user for message to be published to the stream. User can always exit by typing 0
+    var recursiveUserInput = () => {
+      input.question(
+        "Enter your message to publish or Type 0 to exit:\n",
+        (userInput) => {
+          if (userInput === "0") {
+            producer.close();
+            return input.close();
+          }
+
+          const data = {
+            payload: Buffer.from(userInput).toString("base64")
+          };
+          producer.send(JSON.stringify(data));
+          console.log(`Message sent: ${userInput}`);
+          recursiveUserInput();
+        }
+      );
     }
+    recursiveUserInput();
+  });
+  producer.onclose = function () {
+    console.log("Closed WebSocket:Producer connection for " + stream);
+  };
 }
 
-// Subscribe to stream
-async function createConsumer() {
-  const dcList = await getDCList();
-  await console.log("dcList: ", dcList);
-
-  try{
-    console.log("Creating local stream...");
-    await client.createStream("my-stream", true);
-  } catch(e) {
-    await console.log("Stream could not be created because "+ e);
-  }
+async function receiveData () {
+  console.log("\n ------- Receive Messages  ------");
+  const consumer = await client.createStreamReader(
+    stream,
+    "test-subscription-1",
+    is_local
+  );
+  
+  // To close consumer connection when user types 0
```

Comment:
```suggestion
  // Close consumer connection when user types 0
```

Reply:
Yeah, this sort of thing would not make as much sense in the step-by-step. I think it works well here, though.

Reply:
exactly! that's why I am not sure if we should break this exactly into step by step or not..

Or there is another option.. we can keep this good full demo file in just https://macrometa.com/docs/streams/stream-examples/pub-sub-streams

And we can keep just simple hardcoded message "Hello World" in Getting Started page( https://macrometa.com/docs/streams/getting-started-streams)?

Reply:
cc @dallas-macrometa

Reply:
I think the full demo file is better code and will be more useful for regular developers, but I think the step-by-step explanations hold value for less advanced users.

Reply:
yup!

Reply:
We just need to make sure we explain what is going on so that users do not get confused. I suspect that the more advanced folks will skip ahead to the full demo anyway, because that's what devs seem to do.